pkg/kubectl/cmd/apply/apply.go

```diff
@@ -194,14 +195,15 @@ func NewCmdApply(baseName string, f cmdutil.Factory, ioStreams genericclioptions
 func (o *ApplyOptions) Complete(f cmdutil.Factory, cmd *cobra.Command) error {
 	o.ServerSideApply = cmdutil.GetServerSideApplyFlag(cmd)
 	o.ForceConflicts = cmdutil.GetForceConflictsFlag(cmd)
+	o.FieldManager = cmdutil.GetFieldManagerFlag(cmd)
 	o.DryRun = cmdutil.GetDryRunFlag(cmd)
 
 	if o.ForceConflicts && !o.ServerSideApply {
-		return fmt.Errorf("--force-conflicts only works with --server-side")
+		return fmt.Errorf("--experimental-force-conflicts only works with --experimental-server-side")
 	}
 
 	if o.DryRun && o.ServerSideApply {
-		return fmt.Errorf("--dry-run doesn't work with --server-side")
+		return fmt.Errorf("--dry-run doesn't work with --experimental-server-side")
```

Comment:
Why not? Wasn't this most of the point of dry run?

Reply:
`--dry-run` here is the client-side version.

Reply:
I don't think client-side dry-run makes sense with server-side apply, and I think it shouldn't be allowed. I could suggest `did you mean --experimental-server-dry-run`?

Reply:
ah! Please add a "(did you mean --experimental-server-dry-run instead?)" to the message.

Reply:
I think it should be "(did you mean --server-dry-run instead?)"

Reply:
Yes, I noticed before making the change, we should be good ;-)